doi-proxy/src/test/java/no/unit/nva/doi/CrossRefClientTest.java

```diff
@@ -117,18 +135,82 @@ public void fetchDataForDoiReturnAnEmptyOptionalForANonExistingUrl()
 
     @Test
     @DisplayName("fetchDataForDoi returns an empty Optional for an unknown error")
-    public void fetchDataForDoiReturnAnEmptyOptionalForAnUnknownError()
+    void fetchDataForDoiReturnAnEmptyOptionalForAnUnknownError()
             throws URISyntaxException {
         CrossRefClient crossRefClient = crossRefClientReceives500();
         Optional<String> result = crossRefClient.fetchDataForDoi(DOI_STRING).map(MetadataAndContentLocation::getJson);
         assertTrue(result.isEmpty());
     }
 
     @Test
-    public void fetchDataForDoiThrowsExceptionWhenDoiHasInvalidFormat() {
+    void fetchDataForDoiThrowsExceptionWhenDoiHasInvalidFormat() {
         assertThrows(IllegalArgumentException.class, () -> crossRefClient.fetchDataForDoi(ILLEGAL_DOI_STRING));
     }
 
+    @Test
+    void crossrefClientThrowsRuntimeExceptionIfEnvironmentVariableCrossrefApiTokenNameIsMissing() {
+        Environment mockEnvironment = mock(Environment.class);
+        when(mockEnvironment.readEnvOpt(CROSSREFPLUSAPITOKEN_KEY_ENV)).thenReturn(Optional.of("anything"));
+        Executable executable = () -> new CrossRefClient(HttpClient.newHttpClient(), mockEnvironment,
+                new SecretsReader());
+        Exception exception = assertThrows(RuntimeException.class, executable);
+        String actual = exception.getMessage();
+        assertThat(actual, containsString(MISSING_CROSSREF_TOKENS_ERROR_MESSAGE));
```

Comment:
This does not assert what the name of the test implies it should assert, does it?